### ISSUE TITLE
tf.contrib.copy_graph.copy_op_to_graph Error when copying a foldl operation

### ISSUE THREAD

**Issue by EmanueleGhelfi** [tensorflow/tensorflow#24339]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Archlinux
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 1.12
- Python version: 3.6
- CUDA/cuDNN version: cuda 9.0, cudnn 7.1
- GPU model and memory: GeForce GTX 1060 

**Describe the current behavior**
tf.contrib.copy_graph.copy_op_to_graph gives a 
```
RecursionError: maximum recursion depth exceeded while calling a Python object
```
when called on a foldl operation.
tf.contrib.copy_graph.copy_op_to_graph does not manages foldl operation. Maybe it is because of the recursive structure of the operation.

**Describe the expected behavior**
tf.contrib.copy_graph.copy_op_to_graph should copy the part of the graph containing the fold operation. Maybe it can manage in a particular way recursive blocks like fold.

**Code to reproduce the issue**
```python
import tensorflow as tf
x = tf.ones((2,1))
acc =  tf.ones((1,1))
# definition of the fold operation
y = tf.foldl(lambda a,y: return a + y, x, acc) 
# tf.Session().run(y) gives [[3.]], as expected

# copy the fold to another graph
g = tf.Graph()
with g.as_default():
    tf.contrib.copy_graph.copy_op_to_graph(y, g, [])
    # RecursionError: maximum recursion depth exceeded while calling a Python object
```
**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.
Full traceback attached.
[tf_bug.zip](https://github.com/tensorflow/tensorflow/files/2675691/tf_bug.zip)

Basically the issue is a recursion issue, tf does not understands the recursive nature of the foldl op.
Reduced traceback: 
```
RecursionError: maximum recursion depth exceeded while calling a Python object
```

**Comment by ymodak** [comment]
@alextp Can you PTAL? Thanks!

**Comment by alextp** [comment]
Contrib is deprecated and about to be deleted so I don't have the bandwidth to investigate this.

I also suspect that with control flow v2 this won't be a problem.
